### DISCUSSION TITLE
Remote Control Option

### DISCUSSION THREAD

**Discussion by Autperformance** [pear-devs/pear-desktop#1566] [Ideas]
So i was searching for a way to display music names etc. through a website (browser source to obs) and went with youtube music app (not this one) - everything worked like it should BUT it has ads - alright so i searched for a non-ad version and found this one.

problem with this one is it has no remote control option (or at least i couldnt find it in the settings/plugins etc.) - searched in the github, found post from 2021 but i guess it never got implemented?!

now i have an app with ads and remote control or no ads no remote control. both are not a thing (yet?)

please let me know if i missed the plugin or if there will be one, thank you! 

first time using github btw, if i posted this wrong let me know, thanks.

**Comment by inson1** [comment]
@Autperformance Wouldnt be enought Tuna-OBS: Integration with [OBS](https://obsproject.com/)'s plugin [Tuna](https://obsproject.com/forum/resources/tuna.843/)? sorry if not, I never used it

**Reply by Autperformance** [reply]
hmm i dont know what "Tuna-OBS" does - i just know that the website i want to use for the obs "now playing" browser source (you can find it here if you want to take a look at it: https://docs.6klabs.com/docs/accounts-apps/youtube-music ) needs that "remote control" option to be turned on and i dont know if there is another way to connect the website to the youube music app without the remote control - havent found anything yet.

**Reply by Destroy666x** [reply]
You seem to be lost in approaching the problem in the 1st place. Is your goal displaying song info in OBS? If not, specify the exact goal, please. Within user terms as to what is the desired outcome, not as XY type of problem with bunch of mentions of potential solutions.

If yes, then the above solution is exactly what you need. Here's an example of a dynamic widget I made for my stream:
![image](https://github.com/th-ch/youtube-music/assets/3606072/8897e447-acce-48c1-851c-5fff54b9de1d)
- Tuna plugin is used for song info and progress bar. The plugin page has good documentation as to how to set it up.
- Waveform for the graph
- and Advanced Scene Switcher for hiding the widget if nothing is playing and adding some obs-shaderfilter effects when the song changes
I might create a YT tutorial vid for this setup this or next month.

**Reply by Tenelux01** [reply]
Im glad i decided to check the discusions. I would like to put my vote for the remote control. Im in the same sitch i cant find a program that has both. Sure tuna works but all i get is a progress bar. I cant figure it out lol, nothing is pulled or created, forum after forum i see my settings are correct. I went to the directories the lyrics and covers are supposed to be and nothing is there. And yes, amuse does the same thing as tuna its just a lot easier. I dont think Autperformance is confused on what they want to do i think theres unawareness that tuna is an alternative. In short, amuze is better and easier ive tested it with the remote option in an alt version, tuna imo is just bad. Where do i sign for the remote control

**Reply by JellyBrick** [reply]
See #1909

**Reply by Tenelux01** [reply]
unless my settings are configed differently then the api doesnt work for it, yet. i thought it was a way to but it wasnt, ive tried even without a 1st instance authorization and nothin. if the numbers are supposed to be something else then idk what they should be

**Comment by QuditWolf** [comment]
There is some discussion and PRs on addition of an api(&browser) based metadata/RPC and song control.

Have a look at my PR :D

https://github.com/th-ch/youtube-music/pull/1151

It's been marked deprecated though (I last pushed it ~6mo back... but I still use it) I'll try merging the new main with it.

**Reply by Autperformance** [reply]
i just read your post and its a bit complicated for me like really a lot to take in - im not a pc expert at all, i was just searching for this feature which this program doesnt have. So you say with your "thing" (i dont even know what it is, like code? plugin? i can do what i described? how would i even install that or add it to the program, is there a step by step tutorial or anything like that?
